Cookbook/Proposals/MealyReactiveFeedback.md

```diff
@@ -0,0 +1,141 @@
+The ideal approach will be to **separate "external events" as plain `Signal<Event>`, and pass it as a new argument next to `Feedback`s**.
+
+The idea of "external event" has been previously discussed in [Babylonpartners/ReactiveFeedback#32](https://github.com/Babylonpartners/ReactiveFeedback/pull/32), and it also makes sense from automata theory that "external events" (alphabets) and "feedbacks" (output function) are different.
+Also, Elm provides [`Subscription`](https://package.elm-lang.org/packages/elm/core/latest/Platform-Sub) which works just as same as observing "external events" and it is different from feedback type called `Cmd<Event>`.
+
+Furthermore, by separating "external event" nicely, we can send "initial external event" only after system is instantiated, so that the racing issue described in [ReactiveFeedback/pull#38](https://github.com/Babylonpartners/ReactiveFeedback/pull/38#issuecomment-468478325) will no longer occur.
+
+### 3. Remove duplicated `state` caching in both `SignalProducer` and `Property`
+
+`State` is stored in both `scan` and `Property` which is obviously inefficient.
+
+## Proposed solution
+
+### Changes
+
+```diff
+ public struct Feedback<State, Event> {
+-    let events: (Scheduler, Signal<State, NoError>) -> Signal<Event, NoError>
++    let events: (Scheduler, Signal<(Event, State), NoError>) -> Signal<Event, NoError>
+ }
+
+ extension Property {
+     public convenience init<Event>(
+         initial: Value,
+         scheduler: Scheduler = QueueScheduler.main,
+         reduce: @escaping (Value, Event) -> Value,
++        events: Signal<Event>,   // external events
+         feedbacks: [Feedback<Value, Event>]
+     ) {
+         ...
+     }
+ }
+```
+
+- `Feedback` preserves the streaming format, i.e. `Signal<???> -> Signal<Event>` rather than `??? -> Signal<Event>` so that it can have flattening capability e.g. `flatMap(.latest)` or `flatMap(.first)`.
+- Resolves [Babylonpartners/ReactiveFeedback#32](https://github.com/Babylonpartners/ReactiveFeedback/pull/32)
+- Improves [Babylonpartners/ios-playbook#44](https://github.com/Babylonpartners/ios-playbook/pull/44)
+    - Resolves "idle -> routing -> idle" navigation state transition dance.
+- **Backward compatible** with current Moore machine (we can still add as many states for precise control as we want)
+
+### Usage
+
+```swift
+let externalEvent = Signal.merge(
+    increment.map { Event.increment },
+    decrement.map { Event.decrement }
+)
+
+let system = Property(
+    initial: 0,
+    reduce: counterReducer,
+    events: externalEvent,
+    feedbacks: [loggingFeedback, analyticsFeedback]
+)
+
+/// We can simply create SignalProducer from this system,
+/// without creating producer first then bind to property.
+let p = system.producer
+```
+
+## Impact on existing codebase
+
+This proposal will be a **breaking change**, but modification will be relatively **small** because current state structures can be fully preserved (though it can be minimized).
```

Comment:
Not sure why is it a breaking change?

Reply:
It's because `Property.init` type signature changes.